crates/iroha_executor/src/default/isi/multisig/transaction.rs

```diff
@@ -0,0 +1,249 @@
+//! Validation and execution logic of instructions for multisig transactions
+
+use alloc::collections::{btree_map::BTreeMap, btree_set::BTreeSet};
+
+use super::*;
+
+impl VisitExecute for MultisigPropose {
+    fn visit<V: Execute + Visit + ?Sized>(&self, executor: &mut V) {
+        let proposer = executor.context().authority.clone();
+        let target_account = self.account.clone();
+        let host = executor.host();
+        let instructions_hash = HashOf::new(&self.instructions);
+        let multisig_role = multisig_role_for(&target_account);
+        let is_downward_proposal = host
+            .query_single(FindAccountMetadata::new(
+                proposer.clone(),
+                SIGNATORIES.parse().unwrap(),
+            ))
+            .map_or(false, |proposer_signatories| {
+                proposer_signatories
+                    .try_into_any::<BTreeMap<AccountId, u8>>()
+                    .dbg_unwrap()
+                    .contains_key(&target_account)
+            });
+        let has_multisig_role = host
+            .query(FindRolesByAccountId::new(proposer))
+            .filter_with(|role_id| role_id.eq(multisig_role))
+            .execute_single()
+            .is_ok();
+
+        if !(is_downward_proposal || has_multisig_role) {
+            deny!(executor, "not qualified to propose multisig");
+        };
+
+        if host
+            .query_single(FindAccountMetadata::new(
+                target_account.clone(),
+                approvals_key(&instructions_hash),
+            ))
+            .is_ok()
+        {
+            deny!(executor, "multisig proposal duplicates")
+        };
+    }
+
+    fn execute<V: Execute + Visit + ?Sized>(self, executor: &mut V) -> Result<(), ValidationFail> {
+        let proposer = executor.context().authority.clone();
+        let target_account = self.account;
+
+        // Authorize as the multisig account
+        executor.context_mut().authority = target_account.clone();
```

Comment:
why do this here. Shouldn't the account executing this instruction have the multisig role?